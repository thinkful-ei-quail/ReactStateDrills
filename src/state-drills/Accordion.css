ul {
  list-style: none;
}

li {
  background-color: lightblue;
  width: 500px;
  border-style: solid;
}

button {
  width: 100%;
  background-color:#666666;
  color: white;
}

p {
  padding-left: 10px;
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
}