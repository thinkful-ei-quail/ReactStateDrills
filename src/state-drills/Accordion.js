import React from 'react';

class Accordion extends React.Component {
static defaultProps = {sections:[]}
  state = {
    currentIndex: 0
  }

  handleButtonClicked = (index) => {
    this.setState({currentIndex: index});
  };

  renderButtons() {
    return this.props.sections.map((section, index) => (
      <li key={index}><button onClick={() => this.handleButtonClicked(index)}>
        {section.title}
      </button>
      {index === this.state.currentIndex ? <p>{section.content}</p>: ''}
      </li>
    ))
  }

  render() {
    return (
      <div>
        <ul>
          {this.renderButtons()}
        </ul>
      </div>
    )
  }

}

export default Accordion