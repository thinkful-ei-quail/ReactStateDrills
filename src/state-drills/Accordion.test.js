import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion'

describe(`Accordion Component`, () => {

  const sections = [
    { name: 'First tab',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.' },
    { name: 'Second tab',
      content: 'Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit.' },
    { name: 'Third tab',
      content: 'Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur.' },
  ];

  //smoke test
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  //snapshot test
  it('renders empty given no sections', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the first section by default', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  //event test
  it('closes the first section and opens any clicked section', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})