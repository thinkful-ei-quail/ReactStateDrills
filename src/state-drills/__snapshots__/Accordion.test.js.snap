// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component closes the first section and opens any clicked section 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      />
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      />
      <p>
        Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      />
    </li>
  </ul>
</div>
`;

exports[`Accordion Component renders empty given no sections 1`] = `
<div>
  <ul />
</div>
`;

exports[`Accordion Component renders the first section by default 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      />
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.
      </p>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      />
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      />
    </li>
  </ul>
</div>
`;
